+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -749,12 +749,16 @@ function create_solver(obj, solver_options, plugin, regenerate)
             end
             solver_options.assume_lower_bounds = true;
 
-            if regenerate || isempty(obj.solver) || (nosnoc.solver.MpccMethod(plugin) ~= obj.solver.relaxation_type)
+            if regenerate || isempty(obj.solver)
                 obj.solver = nosnoc.solver.mpccsol('Mpcc solver', plugin, obj, solver_options);
             end
         end
 
-        function stats = solve(obj)
+        function stats = solve(obj, active_set)
+            arguments
+                obj
+                active_set {mustBeA(active_set,["nosnoc.activeset.Pds","double"])} = nosnoc.activeset.Pds.empty;
```

Comment:
oh boy, what is this. please add one line comments,

Reply:
I mean I think it is pretty readable: `mustBeA` does exactly what it seems: it checks if active_set is either an active set or a double (which is necessary to allow passing `[]` because matlab treats empties of different classes differently and has no concept of e.g. `null`) though I can actually make this more readable by making it:

`mustBeA(active_set, ["nosnoc.activeset.Base"])`

and having `nosnoc.activeset.Base.empty` be the default active set.

Reply:
Also to explain why you need to use the `mustBeA` verification function rather than passing a type is because I suspect Mathworks only heard of how programming languages are built and decided that type-coersion based on the arguments block is a sane default if you do e.g.:

```MATLAB
function foo = fun(foo)
arguments
   foo SomeClass
end
end
```

and call `x = foo([])` you will _not_ get `x = []` if `SomeClass` has a constructor that takes exactly one argument, you will get `x = SomeClass([])`. Which is _insane_.

Reply:
Thank you for the explanation. One-two line comment in code is appreciated.